js/filter.js

```diff
@@ -3,23 +3,98 @@
 (function () {
 
   var MAX_PINS_ON_MAP = 5;
-  var valueOfAllAds = 'any';
+  var VALUE_OF_ALL_ADS = 'any';
+  var PRICE_MAP = {
+    'any': {
+      min: -Infinity,
+      max: Infinity
+    },
+    'middle': {
+      min: 10000,
+      max: 50000
+    },
+    'low': {
+      min: 0,
+      max: 10000
+    },
+    'high': {
+      min: 50000,
+      max: Infinity
+    }
+  };
+
+  var filterNode = document.querySelector('.map__filters');
+  var typeFilterNode = filterNode.querySelector('#housing-type');
+  var priceFilterNode = filterNode.querySelector('#housing-price');
+  var roomsFilterNode = filterNode.querySelector('#housing-rooms');
+  var guestsFilterNode = filterNode.querySelector('#housing-guests');
+  var featureFilterNodesArr = Array.from(filterNode.querySelectorAll('.map__checkbox'));
+
+  var filterPinsOfType = function (dataElement) {
+    if (typeFilterNode.value === dataElement.offer.type || typeFilterNode.value === VALUE_OF_ALL_ADS) {
+      return true;
+    }
+
+    return false;
+  };
+
+  var filterPinsOfPrice = function (dataElement) {
+    if (PRICE_MAP[priceFilterNode.value].min < dataElement.offer.price && PRICE_MAP[priceFilterNode.value].max > dataElement.offer.price) {
+      return true;
+    }
+
+    return false;
+  };
```

Comment:
- Давай облегчим фильтры:
```suggestion
  var filterPinsByPrice = function (dataElement) {
    return PRICE_MAP[priceFilterNode.value].min < dataElement.offer.price && PRICE_MAP[priceFilterNode.value].max > dataElement.offer.price;
  };
```
- Правильнее писать `filter by` а не `filter of`